packages/amazonq/src/inlineChat/output/computeDiff.ts

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { type LinesOptions, diffLines, Change } from 'diff'
+import * as vscode from 'vscode'
+import { InlineTask, TextDiff } from '../controller/inlineTask'
+
+export function computeDiff(response: string, inlineTask: InlineTask, isPartialDiff: boolean): TextDiff[] | undefined {
+    if (!response) {
+        return
+    }
+    const selectedRange = inlineTask.selectedRange
+    const partialSelectedText = inlineTask.partialSelectedText ?? ''
+    const selectedText = isPartialDiff ? partialSelectedText : inlineTask.selectedText
+
+    const normalizedResponse =
+        getLeadingWhitespace(selectedText) + response.trim() + getTrailingWhitespace(selectedText)
+
+    const diffs = diffLines(selectedText, normalizedResponse, {
+        stripTrailingCr: true,
+        ignoreNewlineAtEof: true,
+    } as LinesOptions)
+
+    const textDiff: TextDiff[] = []
+    let startLine = selectedRange.start.line
+
+    diffs.forEach((part: Change) => {
+        const count = part.count ?? 0
+        if (part.removed) {
+            if (part.value !== '\n') {
+                textDiff.push({
+                    type: 'deletion',
+                    originalText: part.value,
+                    range: new vscode.Range(startLine, 0, startLine + count, 0),
+                })
+            }
+        } else if (part.added) {
+            if (part.value !== '\n') {
```

Comment:
can you double check if all '\n' work in Windows?

Reply:
This is checking the diff result which strips `\r` so this should be fine on windows.